libopendavinci/src/odcore/data/Container.cpp

```diff
@@ -38,13 +38,15 @@ namespace odcore {
                 m_dataType(UNDEFINEDDATA),
                 m_serializedData(),
                 m_sent(TimeStamp(0, 0)),
-                m_received(TimeStamp(0, 0)) {}
+                m_received(TimeStamp(0, 0)),
+                m_sampleTimeStamp(TimeStamp(0, 0)) {}
 
         Container::Container(const SerializableData &serializableData) :
                 m_dataType(serializableData.getID()),
                 m_serializedData(),
                 m_sent(TimeStamp(0, 0)),
-                m_received(TimeStamp(0, 0)) {
+                m_received(TimeStamp(0, 0)),
+                m_sampleTimeStamp(TimeStamp(0, 0)) {
```

Comment:
This is the creation phase of the container; all time stamps are set to zero unless specified otherwise.
Shortly before sending, the sampleTimeStamp is set to send time stamp if not set to a value > 0.